### PR TITLE
Documents feature flags in README and add support for lazy-static-context

### PR DESCRIPTION
**Pull request by sorpaas**
This adds new feature `lazy-static-context`, which uses `lazy_static` for context allocation in heap. It's technically mostly the same as disabling both `static-context` and `lazy-static-context`, just that this allows users to use the same interface as `static-context`.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -96,7 +106,7 @@ impl PublicKey {
         PublicKey(p)
     }
 
-    #[cfg(feature = "static-context")]
+    #[cfg(all(feature = "static-context", feature = "lazy-static-context"))]
```
Comment (prestwich) [inline src/lib.rs:109]: I think you want `any` instead of `all`

Comment (prestwich) [reply src/lib.rs:109]: According to the README
>   Only one of `static-context` and `lazy-static-context` can be enabled,

File: src/lib.rs [hunk 2]
```diff
@@ -20,20 +20,30 @@ use arrayref::{array_ref, array_mut_ref};
...
 use typenum::U32;
-#[cfg(feature = "std")]
-use serde::{Serialize, Deserialize, ser::Serializer, de};
 
 use crate::{
     Error, curve::{Scalar, Field, Affine, Jacobian, ECMultContext, ECMultGenContext},
     util::{self, Decoder, SignatureArray},
 };
 
+#[cfg(all(feature = "static-context", feature = "lazy-static-context"))]
```
Comment (prestwich) [inline src/lib.rs:37]: I think you want

```
#[cfg(all(not(feature = "static-context"), feature = "lazy-static-context"))]
```

File: src/lib.rs (hunk 1 above)
Comment (prestwich) [reply src/lib.rs:109]: (this applies throughout this file)

**Comment by LLFourn** [comment]
Just an FYI on `lazy_static`:  https://github.com/rust-lang-nursery/lazy-static.rs/issues/163. Apparently is causes `cargo audit` to fail which might be undesirable for a crypto library.

**Pull request merged**